platform/darwin/src/MGLShape.mm

```diff
@@ -1,7 +1,95 @@
 #import "MGLShape.h"
 
+#import "NSString+MGLAdditions.h"
+#import "MGLTypes.h"
+
+#import <mbgl/util/geo.hpp>
+
+bool compareByCoordinate(CLLocationCoordinate2D lhs, CLLocationCoordinate2D rhs)
+{
+    return lhs.latitude+lhs.longitude < rhs.latitude+rhs.longitude;
+}
+
+bool operator!=(const CLLocationCoordinate2D lhs, const CLLocationCoordinate2D rhs) {
+    return lhs.latitude != rhs.latitude || lhs.longitude != rhs.longitude;
+}
+
+bool operator==(std::vector<CLLocationCoordinate2D>& lhs, std::vector<CLLocationCoordinate2D>& rhs)
+{
+    if (lhs.size() != rhs.size()) return false;
+    
+    std::sort(lhs.begin(), lhs.end(), compareByCoordinate);
```

Comment:
Order matters. Two coincident lines, one of which goes in the opposite direction as the other, are not equal.